tools/PI/DevHome.PI/ViewModels/BarWindowViewModel.cs

```diff
@@ -367,18 +373,25 @@ public void ManageExternalToolsButton_ExternalToolLaunchRequest(object sender, E
         InvokeTool(tool, TargetAppData.Instance.TargetProcess?.Id, TargetAppData.Instance.HWnd);
     }
 
-    private void InvokeTool(ExternalTool tool, int? pid, HWND hWnd)
+    private async void InvokeTool(ExternalTool tool, int? pid, HWND hWnd)
     {
-        var process = tool.Invoke(pid, hWnd);
-        if (process is null)
+        try
+        {
+            var process = await tool.Invoke(pid, hWnd);
+        }
+        catch (Exception ex)
         {
-            // A ContentDialog only renders in the space its parent occupies. Since the parent is a narrow
-            // bar, the dialog doesn't have enough space to render. So, we'll use MessageBox to display errors.
-            Windows.Win32.PInvoke.MessageBox(
-                HWND.Null, // ThisHwnd,
-                string.Format(CultureInfo.CurrentCulture, _errorMessageText, tool.Executable),
-                _errorTitleText,
-                Windows.Win32.UI.WindowsAndMessaging.MESSAGEBOX_STYLE.MB_ICONERROR);
+            _log.Error(ex, "Tool launched failed");
+
+            var builder = new StringBuilder();
+            builder.AppendLine(ex.Message);
```

Comment:
`ex.ToString()` will include the inner exception.  I believe.

Reply:
You're correct it does, but it also includes a callstack, which in this instance isn't useful - this scenario just needs a simple message stating the root cause.